Swap out error_chain with failure_crate and JSONize log output

zv opened the pull request:
Resolves #12 #4 and https://github.com/polyswarm/development/issues/30

mjkoo commented:
Seems like a good number of my comments from https://github.com/polyswarm/relay/pull/12 are still relevant, rather than closing and opening a new PR would appreciate either a resolution or a comment in the review about why the review item wasn't relevant.

zv commented:
@mjkoo I added a new commit that addresses some of the issues you raised in #12 as well as some other issues that were given new consideration. Can you review my changes again?
